### PR TITLE
Add project dropdown search feature

### PR DESCRIPTION
**Pull request by adammw**
:koala:

samson_dropdown

/cc @sandlerr @princemaple @jwswj (can someone cc @zendesk/samson please?)

### CODE REVIEW

File: app/assets/javascripts/directives/filter_by_directive.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+samson.directive('filterBy', function($timeout, $parse) {
```
Comment (princemaple) [inline app/assets/javascripts/directives/filter_by_directive.js]: `$timeout`, evidence of `copy-paste`..

File: app/views/shared/_header.html.erb [hunk 2]
```diff
@@ -6,10 +6,12 @@
     <ul class="nav navbar-nav">
       <% if current_user %>
         <li class="dropdown <%= 'active' if request.path.starts_with?('/projects') %>">
-          <a href="#" class="dropdown-toggle" data-toggle="dropdown">Projects <b class="caret"></b></a>
+          <a href="#" class="dropdown-toggle" data-toggle="dropdown" focus-on="{ click: '#project_search' }">Projects <b class="caret"></b></a>
           <ul class="dropdown-menu">
+            <li><a class="padding-sm"><input id="project_search" type="search" ng-model="projectSearchQuery" ng-trim="false" handle-dropdown-selection class="form-control input-block" placeholder="Search projects"></a></li>
```
Comment (princemaple) [inline app/views/shared/_header.html.erb]: separate lines

File: app/assets/javascripts/directives/handle_dropdown_selection_directive.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+samson.directive('handleDropdownSelection', function($timeout, $parse) {
```
Comment (princemaple) [inline app/assets/javascripts/directives/handle_dropdown_selection_directive.js]: `$parse` is not used

**Comment by princemaple** [comment]
:+1: 

/cc @zendesk/samson

File: app/assets/javascripts/directives/filter_by_directive.js (hunk 1 above)
Comment (adammw) [inline app/assets/javascripts/directives/filter_by_directive.js]: Yeah, I told myself to make sure I fix that before I pushed, but forgot. Fixed now.

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -43,3 +43,13 @@
 a.no-hover:hover {
   text-decoration: none;
 }
+
+.dropdown-menu > li > a.padding-sm {
```
Comment (jwswj) [inline app/assets/stylesheets/application.scss]: I saw this and thought:
```
.dropdown-menu > li {
  > a.padding-sm {
    padding: 3px 10px;
  }
  &.selected > a {
    text-decoration: none;
    color: #262626;
    background-color: #f5f5f5;
  }
}
```
but it just seems more complicated.

**Comment by jwswj** [comment]
:+1: Love it

**Comment by sandlerr** [comment]
:+1: awesome!

**Pull request merged**